test/ICSharpCode.SharpZipLib.Tests/Serialization/SerializationTests.cs

```diff
@@ -0,0 +1,202 @@
+﻿using System;
+using System.IO;
+using System.Runtime.Serialization;
+
+using NUnit.Framework;
+using ICSharpCode.SharpZipLib.BZip2;
+using ICSharpCode.SharpZipLib.Core;
+using ICSharpCode.SharpZipLib.GZip;
+using ICSharpCode.SharpZipLib.Lzw;
+using ICSharpCode.SharpZipLib.Tar;
+using ICSharpCode.SharpZipLib.Zip;
+
+namespace ICSharpCode.SharpZipLib.Tests.Serialization
+{
+	[TestFixture]
+	public class SerializationTests
+	{
+		/// <summary>
+		/// Test that SharpZipBaseException can be serialized.
+		/// </summary>
+		[Test]
+		[Category("Core")]
+		[Category("Serialization")]
+		public void SerializeSharpZipBaseException()
```

Comment:
All of these test (except for`ValueOutOfRangeException`) can be replaced by a parameterized test:
```cs
/// <summary>
/// Test that SharpZipLib Custom Exceptions can be serialized.
/// </summary>
[Test]
[Category("Core")]
[Category("Serialization")]
[TestCase(typeof(SharpZipBaseException))]
[TestCase(typeof(InvalidNameException))]
[TestCase(typeof(StreamDecodingException))]
[TestCase(typeof(UnexpectedEndOfStreamException))]
[TestCase(typeof(StreamUnsupportedException))]
[TestCase(typeof(BZip2Exception))]
[TestCase(typeof(GZipException))]
[TestCase(typeof(LzwException))]
[TestCase(typeof(TarException))]
[TestCase(typeof(ZipException))]
public void SerializeException(Type exceptionType)
{
    string message = $"Serialized {exceptionType.Name}";
    var exception = Activator.CreateInstance(exceptionType, message);

    var deserializedException = ExceptionSerialiseHelper(exception) as Exception;
    Assert.That(deserializedException.Message, Is.EqualTo(message), "deserialized message should match original message");
}
```

Reply:
I can do that if you want.